frappe/templates/base.html

```diff
@@ -65,7 +65,7 @@
 			frappe.ready_events.push(fn);
 		}
 		window.dev_server = {{ dev_server }};
-		window.socketio_port = {{ frappe.socketio_port }};
+		window.socketio_port = {{ (frappe.socketio_port or (3000 if dev_server else 80)) }};
```

Comment:
Doesn't work well on multi-bench setup. I'd say it should be the responsibility of the developer to make sure ports are correctly set in the config.

Reply:
I do agree that the ops team should make sure that ports are defined in the config (though it can be hard to know all available / new configs), but I still believe Frappe should be reliable enough to not generate invalid JavaScript code like `window.socketio_port = None`.

Any suggestion on a compromise ?
Maybe initialize with `frappe.webserver_port` instead ?

Reply:
so, what should we do ?

Reply:
so @netchampfaris , what should we do ?

Reply:
I like 3. 

@surajshetty3416 Thoughts?

Reply:
Yeah, option 3 seems safe... let's go with it.

Reply:
Okay. Done on this PR plus v11 (#10121) and v12 (#10122).

Thanks @netchampfaris & @surajshetty3416 .